server/datastore/mysql/migrations/tables/20240126020643_AddHostActivities.go

```diff
@@ -49,23 +49,23 @@ func Up_20240126020643(tx *sql.Tx) error {
 		return errors.Wrap(err, "create host_activities table")
 	}
 
-	// Prior to this update, the database didn't differentiate between
-	// "async" and "sync" requests. With Fleet v4.44.0, all async requests
-	// will execute regardless of their pending duration. To avoid
-	// unintended execution of old requests upon server upgrade, these are
-	// now marked as "sync", reflecting their original 5-minute execution
-	// limit.
+	// Force an exit_code for scripts that didn't run previous to this update.
+	// In previous releases, scripts were allowed to be queued only for 5
+	// minutes, since we're removing that restriction, any scripts with a
+	// null exit_code would be unexpectedly sent to the host unless we do
+	// this.
 	const setOldScriptsAsSyncStmt = `
             UPDATE host_script_results hsr
             SET
-                sync_request = 1,
+                exit_code = -1,
                 updated_at = hsr.updated_at
             WHERE
-                user_id IS NULL
+	        exit_code IS NULL
+                AND user_id IS NULL
                 AND created_at < CURRENT_TIMESTAMP
```

Comment:
In theory, we might want to factor in the 5 minute to the comparison, but I personally don't think it is worth the trouble.